lib/base-compiler.js

```diff
@@ -407,8 +407,7 @@ class BaseCompiler {
         if (filters.binary && !this.compiler.supportsBinary) {
             delete filters.binary;
         }
-
-        const executeParameters = options.executeParameters || [];
+        const executeParameters = options.executeParameters || executionOptions || [];
```

Comment:
I'm a tad confused about not using the options.executeParameters in this, what's the reason this was needed?

Reply:
Mostly because I could not get options.executeParameters to be properly filled. It's probably because I'm missing something (Probably the sendCompile stuff is off somewhere)